src/cli/rustup_mode.rs

```diff
@@ -136,253 +129,252 @@ pub fn main() -> Result<utils::ExitCode> {
                     }
                 }),
         )
         .subcommand(
-            SubCommand::with_name("dump-testament")
+            Command::new("dump-testament")
                 .about("Dump information about the build")
-                .setting(AppSettings::Hidden), // Not for users, only CI
+                .hide(true), // Not for users, only CI
         )
         .subcommand(
-            SubCommand::with_name("show")
+            Command::new("show")
                 .about("Show the active and installed toolchains or profiles")
                 .after_help(SHOW_HELP)
                 .arg(
                     verbose_arg("Enable verbose output with rustc information for all installed toolchains"),
                 )
-                .setting(AppSettings::VersionlessSubcommands)
                 .setting(AppSettings::DeriveDisplayOrder)
                 .subcommand(
-                    SubCommand::with_name("active-toolchain")
+                    Command::new("active-toolchain")
                         .about("Show the active toolchain")
                         .after_help(SHOW_ACTIVE_TOOLCHAIN_HELP)
                         .arg(
                             verbose_arg("Enable verbose output with rustc information"),
                         ),
                 )
                 .subcommand(
-                    SubCommand::with_name("home")
+                    Command::new("home")
                         .about("Display the computed value of RUSTUP_HOME"),
                 )
-                .subcommand(SubCommand::with_name("profile").about("Show the current profile"))
-                .subcommand(SubCommand::with_name("keys").about("Display the known PGP keys")),
+                .subcommand(Command::new("profile").about("Show the current profile"))
+                .subcommand(Command::new("keys").about("Display the known PGP keys")),
         )
         .subcommand(
-            SubCommand::with_name("install")
+            Command::new("install")
                 .about("Update Rust toolchains")
                 .after_help(INSTALL_HELP)
-                .setting(AppSettings::Hidden) // synonym for 'toolchain install'
+                .hide(true) // synonym for 'toolchain install'
                 .arg(
-                    Arg::with_name("toolchain")
+                    Arg::new("toolchain").action(ArgAction::Append)
                         .help(TOOLCHAIN_ARG_HELP)
                         .required(true)
-                        .multiple(true),
+                        .multiple_values(true),
                 )
                 .arg(
-                    Arg::with_name("profile")
+                    Arg::new("profile")
                         .long("profile")
                         .takes_value(true)
-                        .possible_values(Profile::names())
+                        .value_parser(PossibleValuesParser::new(PossibleValuesParser::new(Profile::names()))).takes_value(true)
```

Comment:
Why do we need nesting? I guess we only need one PossibleValuesParser.

Reply:
Huh, some copy-paste error.